### PR TITLE
Fixes #439 Index listing boxes sort order not respected

### LINKED ISSUE #139: Notifiers In Admin
I think it would be helpful if the notifiers were also in the Admin. This could be helpful for a variety of modules and creating new modules.

### LINKED ISSUE #439: Listing boxes sort order issue
Listing boxes no longer display in the order set in Index Listing.
SHOW_PRODUCT_INFO_MAIN_NEW_PRODUCTS
SHOW_PRODUCT_INFO_MAIN_FEATURED_PRODUCTS
SHOW_PRODUCT_INFO_MAIN_SPECIALS_PRODUCTS
 SHOW_PRODUCT_INFO_MAIN_UPCOMING

There are other sets included in Stock settings.

### PR DESCRIPTION
**Pull request by scottcwilson**

### CODE REVIEW

**Comment by zcwilt** [comment]
@scottcwilson did you mean https://github.com/zencart/zencart/issues/439

**Comment by scottcwilson** [comment]
Ugh yes.  Morning typing. :(

**Comment by drbyte** [comment]
👍

**Comment by zcwilt** [comment]
:+1:

File: includes/templates/template_default/templates/tpl_index_default.php [hunk 1]
```diff
@@ -29,8 +29,20 @@
...
-<?php require($template->get_template_dir($tplVars['listingBox']['formatter']['template'], DIR_WS_TEMPLATE, $current_page_base, 'listingboxes') . '/' . $tplVars['listingBox']['formatter']['template']); ?>
-<?php } ?>
+<?php 
+   function mainBoxSort($a, $b) {
+     if ($a['formatter']['sortMainPage'] == $b['formatter']['sortMainPage'])
+        return 0;
+     if ($a['formatter']['sortMainPage'] < $b['formatter']['sortMainPage'])
+        return -1;
+     return 1;
+   }
```
Comment (drbyte) [inline includes/templates/template_default/templates/tpl_index_default.php]: Note: the preference is to *not* put functions in a template file.

Will probably move this out into a static method of the listingBox class in a future refactoring.

**Comment by scottcwilson** [comment]
Makes sense.  You can assign that ticket to me when you create it.

**Pull request merged**